### ISSUE TITLE
Add the ability to delay download of files with `DataManager.require`

### ISSUE THREAD

**Issue by Cadair** [sunpy/sunpy#7546]
### Describe the feature

Sometimes knowledge of what files are required by a function is conditional on the arguments to that function. It would therefore be useful if `DataManager.require` could optionally not download the data, that step being done in `DataManager.get` instead.

This would prevent the need for workarounds like this:

https://github.com/HinodeXRT/xrtpy/blob/660a539f3d6c7ee1e4596e86d4fdbbed5d74871e/xrtpy/image_correction/remove_lightleak.py#L213-L226

**Comment by wtbarnes** [comment]
This would also be really useful in aiapy, e.g. https://github.com/LM-SAL/aiapy/blob/68b6ce7b3539fc0e18262c69c8e1c199b19e3cb4/aiapy/response/channel.py#L83-L100

**Comment by tanishy7777** [comment]
does https://github.com/sunpy/sunpy/pull/7845 solve this issue? or is there more work to be done?

**Comment by nabobalis** [comment]
Ah yeah, that PR didn't auto close this issue.

Thanks for the heads up.

**Comment by tanishy7777** [comment]
> Thanks for the heads up.

yep np. I think there are more issues that havent been autoclosed. This is probably because when u open a PR if you want to autoclose it if merged you need to say something along the lines of "Fixes #<PR_no>" otherwise it dosent autoclose it.

**Comment by nabobalis** [comment]
> > Thanks for the heads up.
> 
> yep np. I think there are more issues that havent been autoclosed. This is probably because when u open a PR if you want to autoclose it if merged you need to say something along the lines of "Fixes #<PR_no>" otherwise it dosent autoclose it.

If you spot more, do say

**Comment by tanishy7777** [comment]
> If you spot more, do say

yep sure
